File: 206-reverse-linked-list/206-reverse-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */

struct ListNode* reverseList(struct ListNode* head){
    struct ListNode* c;
    struct ListNode* k;
    struct ListNode* nexttt;
    
    if(head == NULL)
        return NULL;
    
    k = head;
    c = head->next;
    head->next = NULL;
    while(c != NULL)
    {
        nexttt = c->next;
        c->next = k;
        k = c;
        c = nexttt;
    }
    head = k;
    return head; 
}

File: 206-reverse-linked-list/206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        if (head == null)
            return null;
        ListNode prev = head;
        ListNode curr = head.next;
        ListNode tmp;
        
        while(curr != null)
        {
            tmp = curr.next;
            curr.next = prev;
            if (prev == head)
                prev.next = null;
            prev = curr;
            curr = tmp;
        }
        return prev;
    }
}